editor/plugins/version_control_editor_plugin.cpp

```diff
@@ -462,7 +486,9 @@ VersionControlEditorPlugin::VersionControlEditorPlugin() {
 	commit_message->set_custom_minimum_size(Size2(200, 100));
 	commit_message->set_wrap_enabled(true);
 	commit_message->connect("text_changed", callable_mp(this, &VersionControlEditorPlugin::_update_commit_button));
+	commit_message->connect("gui_input", callable_mp(this, &VersionControlEditorPlugin::_commit_message_gui_input));
 	commit_box_vbc->add_child(commit_message);
+	ED_SHORTCUT("version_control/commit", TTR("Commit"), KEY_MASK_CMD | KEY_ENTER);
```

Comment:
Shouldn't this be on `commit_button` (with `set_shortcut()` IIRC)?

Reply:
Note that this directly affects `TextEdit` input. Implementing the same behavior of the "Commit" button would lead to modification of in `_send_commit_msg()`. This will lead to workflow compatibility breakage in 3.x.

Currently, if no files are staged, then the informational hint below the button will display "No files added to stage" message when you press the "Commit" button, preventing the commit. In contrast, this PR at its state allows to commit immediately when typing out commit message, so that pressing <kbd>Ctrl + Enter</kbd> will stage all files automatically for you unless some files were already staged.

In other words, it may end up unexpected for existing users that pressing "Commit" button no longer prevents committing. Some users might still expect and want to stage all files (or some files) themselves manually rather than automatically before committing.

Note that I'm talking primarily in relation to 3.x, where I worked on this initially. The current approach is fairly risk-free, so regressions are unlikely to happen.